### PR TITLE
tests: mem protection new test inherit resource pool

### PR DESCRIPTION
**Pull request by maksimmasalski**
Add new test test_inherit_resource_pool() that verifies a child thread inherits resource pool
assignment of their parent.

I didn't find that test in source code, but documentation describes that action. I think, it will be nice if documentation will be proved by a tests.

Signed-off-by: Maksim Masalski <maksim.masalski@intel.com>

### CODE REVIEW

**Comment by maksimmasalski** [comment]
@nashif @andrewboie @andyross I'm looking for code review of that PR.

**Review by sjg20 — APPROVED** [review]

File: tests/kernel/mem_protect/mem_protect/src/mem_protect.h [hunk 1]
```diff
@@ -30,6 +30,15 @@ static inline void set_fault_valid(bool valid)
 #define MSG_Q_SIZE (10)
 #define MSG_Q_MAX_NUM_MSGS (10)
 #define MSG_Q_ALIGN (2)
+#define PRIORITY 5
+#define STACK_SIZE (512 + CONFIG_TEST_EXTRA_STACKSIZE)
+#define BLK_SIZE_MIN 16
+#define BLK_SIZE_MAX 64
+#define BLK_NUM_MIN 8
```
Comment (andrewboie) [inline tests/kernel/mem_protect/mem_protect/src/mem_protect.h:37]: why are these values introduced? what uses them?

**Review by LeiW000 — APPROVED** [review]
It looks good to me.

File: tests/kernel/mem_protect/mem_protect/src/mem_protect.h (hunk 1 above)
Comment (maksimmasalski) [reply tests/kernel/mem_protect/mem_protect/src/mem_protect.h:37]: In tests/kernel/mem_protect/mem_protect/src/inherit.c 
`K_MEM_POOL_DEFINE(res_pool, BLK_SIZE_MIN, BLK_SIZE_MAX, BLK_NUM_MAX, BLK_ALIGN);`
parent_thr and child_thr during k_thread_create use PRIORITY and STACK_SIZE

**Pull request merged**